### ISSUE TITLE
Option to disable public group browsing

### ISSUE THREAD

**Issue by krille-chan** [krille-chan/fluffychat#243]
Hi there, would it be possible to add an option, (preferably with an optional password for disabling it) to disable browsing public Matrix groups on the homeserver? I would like to move most of my contacts to Matrix, but some of them need a light parental control option and/or the option of disable NSFW stuff from appearing, so some of the public groups that exist on most homeservers are a problem in that regard. Of course somebody could still find ways to get into those groups if they wanted to, but it a simple hide button for public groups would be enough for most of the users I'm trying to switch, and it should be easy to implement. Thanks for considering this!

**Comment by RasheedAZ** [comment]
Yes please :) this is desperately needed. I came here looking specifically for this kind of feature. I've been thinking of onboarding students to Matrix but it's difficult to find a safe-for-kids client that's a bit feature-complete.

For example in Element, NSFW rooms come up even if you simply click "rooms" in the search bar even when the "Show NSFW content" is toggled off in Settings > Preferences > Room directory!

FluffyChat is better as nothing comes up unless you type something. However, when you do type something in search it shows NSFW rooms and **profiles**.

Idk, but is it possible to have a toggle in advanced settings to disable search altogether... so like email clients you need to know the exact address to communicate with a person or group? That way it'll be relatively safer for kids and we can onboard a new generation of Matrix users hopefully.

**Comment by RasheedAZ** [comment]
Hi. is there a possibility of having this feature on the roadmap for the future? Thanks.

**Comment by mortelil** [comment]
Hey! I also really want this feature. I'm setting up fluffychat for my young daughter to let her learn about computer use and send messages to me and her mother. She's way to curious, and the search bar and ability to add other users or join groups is an issue. Would be great to have some basic parental controls to disable features she doesn't need to be able to chat with her parents.

**Comment by RasheedAZ** [comment]
@mortelil It will presumably take time to implement this in FluffyChat, so you can also try [DeltaChat](https://delta.chat/en/) for small trusted groups for family/friends. It's got E2EE chats and calls, easiest onboarding I've seen, easy multi-device and it's impossible to just browse/search and discover other people's profiles in-app - you need to scan QR codes. Plus it's available on all major platforms and OSes - I use it on a Linux phone and my kids on Linux desktop.
